packages/browser/src/browser/__tests__/integration.test.ts

```diff
@@ -670,10 +671,19 @@ describe('addDestinationMiddleware', () => {
   })
 
   it('supports registering destination middlewares', async () => {
```

Comment:
Might makes sense to break this into a specific test case instead of overloading this test case with this scenario -- 
otherwise, it's easy for this test case // bug to get refactored away. 

Something like: "addDestinationMiddleware should include action destinations" etc. I've seen some test cases that actually include words like 'this tests a bug where XYZ'